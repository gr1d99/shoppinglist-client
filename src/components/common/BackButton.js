import React from 'react';

export const backButton = WrappedClass => {
    return class ShowBackButton extends React.Component {
        render() {
            return (
                <div className="col-md-12">
                    <button className="btn btn-default btn-xs pull-left" onClick={this.props.history.goBack}><span className="glyphicon glyphicon-arrow-left"></span> Go back</button>
                    <WrappedClass {...this.props}/>
                </div>
            )
        }
    }
};