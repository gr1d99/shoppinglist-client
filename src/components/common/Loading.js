import React from 'react';

const Loading = () => {
    /* Helper component to indicate
     * request is running */
    return (
        <div className="loading">
            <p>Loading...</p>
        </div>
    )
};

export default Loading;
