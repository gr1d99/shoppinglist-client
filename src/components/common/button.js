import React from 'react';


const SubmitButton = (props) => {
    return (
      <button
          type={props.type}
          className={props.className}>

          {props.value}

          </button>
    );
};


export default SubmitButton;