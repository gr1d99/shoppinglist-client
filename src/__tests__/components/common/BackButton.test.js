import React from 'react';
import renderer from 'react-test-renderer';
import { backButton } from '../../../src/components/common/BackButton';


describe('BackButton component', () => {
    // back button should render correctly
    it('should render correctly', () => {
        const component = renderer.create(
            <backButton>Buttonq</backButton>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});