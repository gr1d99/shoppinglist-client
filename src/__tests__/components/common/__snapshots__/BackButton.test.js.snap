// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component should render correctly 1`] = `
<backButton>
  Buttonq
</backButton>
`;
